### ISSUE TITLE
Official releases should should use %APPDATA%.

### ISSUE THREAD

**Issue by kaeza** [minetest/minetest#4931]
Up to 0.4.14, the official installers install to `C:\Program Files` which is not a writable location, so the game crashes on Windows 10 when writing `debug.txt` or `minetest.conf`.

All user files should be written to `C:\Users\user\AppData\Roaming\Minetest` (`%APPDATA%\Minetest`) in the same way that Linux builds use `~/.minetest`.

See https://github.com/minetest/minetest/issues/4528#issuecomment-267906193 for more info.

**Comment by Zeno-** [comment]
I think that just to be sure that additionally the working directory be changed to the same location.

**Comment by kaeza** [comment]
If this is done the correct way, you should be able to run Minetest from wherever and it should not affect that directory.

Currently the only file that Minetest insists on writing to the working directory is `debug.txt`. IMHO, that should be changed to be written to `$path_user` in all platforms too. Minetest has a `--logfile` and `--config` options if the user needs to run multiple servers.

**Comment by Zeno-** [comment]
@kaeza good point.

**Comment by nerzhul** [comment]
i also agree, programfiles for engine files, appdata for user mods, texture packs etc, not coming from our release and all written files

**Comment by neuhaus** [comment]
On a shared system you really don't want the directory with the executables to be writable by everyone. It's insecure.

You wouldn't want to write a logfile to `/usr/bin` on Linux either...

**Comment by Zeno-** [comment]
On Linux it's not written to /usr/bin. It's written to the directory that minetest was executed from (the same as for Windows).

**Comment by adrido** [comment]
Oh yes please! that would be relay great.! 😄 

But its not so simple. The portable release should still use the current behavior. Just take a look at that discussion where i told that MinetestMappgerGui isn't designed to work as portable application.: https://forum.minetest.net/viewtopic.php?f=14&t=12139&sid=54ddda2e9eb3263f4f2a0f3792b0b861&start=50#p235902

The problem of the current official installer is, that it installs a portable aka "run_in_place" version that crashes in any modern Windows OS (since Windows Vista). If the Installer would Install a non portable version aka "run_in_place" disabled  it works fine. Just like I have done with my inofficial installers. https://forum.minetest.net/viewtopic.php?f=42&t=14109 

So Minetest still supports this, but I don't know why the Installer installs a portable version of Minetest. That's just stupid. :-/

%Appdata% is imo ok for config or debug.txt. But I would prefer something like ...\<User>\Documents\My Games\Minetest\ to store mods worlds and games. 
And yes I mean "Minetest" not "minetest" or ".minetest" 

I would create a new Installer that does install a correct version of Minetest, but Im currently still unable to create a valid Minetest build. https://github.com/minetest/minetest/issues/4697

**Comment by neuhaus** [comment]
There is quite a bit of discussion to be found regarding use of `%APPDATA%` versus `%UserProfile%/Documents/My Games`.

If interested have a look at
https://blogs.msdn.microsoft.com/oldnewthing/20050701-09/?p=35133/ and
https://blogs.msdn.microsoft.com/oldnewthing/20061228-02/?p=28563

Unfortunately I have not yet found official recommendations by Microsoft. That's probably why it is such a mess.

**Comment by adrido** [comment]
Hmm according to your links the folder should be %UserProfile%/Documents/My Games because:

- The user might be able to paste Games or Mods or Worlds
- The user might zip a world/Game/Mod and send per email/ upload /whatever
- The user might delete or rename worlds
- The user might copy a world and paste them under a different name
- If the User creates a new wold he expect it its saved somwhere

Reasons why it should in %Appdata%:

- The user should not rename/remove Mods or Games because it breaks Mods or worlds
- The user should not change game.conf or mods *.lua files, because he might break the Game /Mod

**Comment by rubenwardy** [comment]
> Up to 0.4.14, the official installers install to C:\Program Files which is not a writable location

False, it defaults to  c://games

**Comment by neuhaus** [comment]
It's true but it's a bad default and could also be considered a bug.

**Comment by dsohler** [comment]
Yes, exactly! Always and 100 percent separate user data and program data.

1. User game data (i.e. the worlds) goes to `%USERPROFILE%\Saved Games\Minetest\`
2. User-related mods/configuration/etc. goes to `%APPDATA%\Minetest\`
3. The installation itself goes to `%PROGRAMFILES%\Minetest\`
4. Built-in mods and configuration goes to `%ALLUSERSAPPDATA%\Minetest\`

I know that Minetest is simply not capable of that, but that would be the correct way according to the official descriptions of said paths.

https://technet.microsoft.com/en-us/library/cc749104(v=ws.10).aspx

USERPROFILE: The user's profile folder.
APPDATA: The file-system directory that serves as a common repository for application-specific data. 
PROGRAMFILES: The Program Files folder.
ALLUSERSAPPDATA: The file-system directory that contains application data for all users.

**Comment by neuhaus** [comment]
You're right, `%USERPROFILE%\Saved Games\` appears to be a standard whereas `%USERPROFILE%/Document/My Games/` is not.

Source: Mention of `FOLDERID_SavedGames`at https://msdn.microsoft.com/en-us/library/windows/desktop/dd378457(v=vs.85).aspx

**Comment by Lawful-Lazy** [comment]
@adrido, seems like a decision between Open Platform and Walled Garden.

I believe that minetest falls into the Open Platform philosophy.

**Comment by adrido** [comment]
>@adrido, seems like a decision between Open Platform and Walled Garden.
>
>I believe that minetest falls into the Open Platform philosophy.

Uhm sorry, I didnt got the point. What do you want to say?

**Comment by Lawful-Lazy** [comment]
By using `%USERPROFILE%\Documents\My Games`, it enables the users to modify their game as they please. This would be an Open Platform.

By using `%APPDATA%`, it discourages users to modify their game. While it's more restrictive, there is less room for error. This is often called a Walled Garden.

I believe that Minetest strives for an Open Platform, so I propose we use `%USERPROFILE%\Documents\My Games`.

**Comment by adrido** [comment]
ahh yes, thats what I tried to say too.

**Comment by red-001** [comment]
`APPDATA` is the correct way to do this on windows, the data folder is also hidden on linux and it's way easier to access the folder on windows. Just tell users to go to `%appdata%\minetest`.

**Comment by red-001** [comment]
Also %ALLUSERSPROFILE% is has been replaced by %PROGRAMDATA% in windows vista.

**Comment by dsohler** [comment]
On Windows 7 `%ALLUSERSPROFILE%` still resolves. It resolves to the same location as `%PROGRAMDATA%` but it wasn't replaced.

**Comment by red-001** [comment]
What about windows 8 and 10?

**Comment by neuhaus** [comment]
This small but **blocking** issue has now been unfixed since 50 million seconds (September 2016).

**Comment by rubenwardy** [comment]
It's not blocking as we don't have an installer

**Comment by neuhaus** [comment]
That's a bad default.

You wouldn't want to install in /games on Linux either, would you?

**Comment by rubenwardy** [comment]
This makes sense to me, but I'd like an official installer (finally) first

**Comment by sfan5** [comment]
they do now: #16129
